### PR TITLE
Fix/46

### LINKED ISSUE #46: Avoid passing service messages (e.g. generator end) to boxes

### PR DESCRIPTION
**Pull request by jslovan**
Fixes #46 . 

Instead of service messages the event `generation_finished` is emitted.  The hierarchy of Message classes has been simplified.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
 	"name": "bakeryjs",
-	"version": "0.0.3",
+	"version": "0.0.4",
```
Comment (jnv) [inline package.json:3]: Please don't do version bumps in PRs, it messes up `npm version` command.

File: src/lib/bakeryjs/Box.ts [hunk 2]
```diff
@@ -332,8 +332,18 @@ abstract class Box extends EventEmitter implements BoxInterface {
 			);
 
 			guardedQ.revoke();
-			this.queue.push(value.createSentinel(siblingsCount, retValue));
-			return;
+			this.emit('generation_finished', [
+				Object.create(null, {
```
Comment (jnv) [inline src/lib/bakeryjs/Box.ts:336]: Why do you need Object.create? Why not just pass an empty object?

File: src/lib/bakeryjs/Message.ts [hunk 3]
```diff
@@ -74,68 +38,32 @@ export type Message = IdMessage | IdSentinel;
...
-		return (this.parent ? `${this.parent.id}` : '') + '/' + this._id;
-	}
+	protected data: MessageData;
+	public readonly parent: Message | undefined;
 
-	protected constructor(parent: IdMessage | undefined) {
+	public constructor(initData?: MessageData, parent?: Message) {
 		this._id = `${messageId++}`;
 		this.parent = parent;
+		this.data = initData ? initData : Object.create(null);
```
Comment (jnv) [inline src/lib/bakeryjs/Message.ts:53]: Again here, why not just use `{}`?

**Review by jnv — CHANGES_REQUESTED** [review]
LGTM, just use of `Object.create` seems unnecessary to me. However, please remove version bump from the PR (package and package-lock)

File: package.json (hunk 1 above)
Comment (jslovan) [reply package.json]: Reverted.

File: src/lib/bakeryjs/Box.ts (hunk 2 above)
Comment (jslovan) [reply src/lib/bakeryjs/Box.ts]: Fixed.

File: src/lib/bakeryjs/Message.ts (hunk 3 above)
Comment (jslovan) [reply src/lib/bakeryjs/Message.ts]: Fixed.

**Review by jnv — APPROVED** [review]

**Pull request merged**